site/src/examples/date-picker/SingleGridPanel.tsx

```diff
@@ -0,0 +1,105 @@
+import { FormField, FormFieldLabel, Input, StackLayout } from "@salt-ds/core";
+import type { DateFrameworkType } from "@salt-ds/date-adapters";
+import {
+  type DateInputSingleDetails,
+  DatePicker,
+  DatePickerOverlay,
+  DatePickerSingleGridPanel,
+  type DatePickerSingleGridPanelProps,
+  DatePickerSingleInput,
+  DatePickerTrigger,
+  type SingleDateSelection,
+  useLocalization,
+} from "@salt-ds/lab";
+import {
+  type ChangeEvent,
+  type SyntheticEvent,
+  useCallback,
+  useState,
+} from "react";
+
+export const SingleGridPanel = () => {
+  const { dateAdapter } = useLocalization();
+  const [numberOfVisibleMonths, setNumberOfVisibleMonths] = useState("1");
+  const [columns, setColumns] = useState("1");
+  const [step, setStep] = useState("1");
+
+  const handleSelectionChange = useCallback(
+    (
+      event: SyntheticEvent,
+      date: SingleDateSelection<DateFrameworkType> | null,
+      details: DateInputSingleDetails | undefined,
+    ) => {
+      const { value, errors } = details || {};
+      console.log(
+        `Selected date: ${dateAdapter.isValid(date) ? dateAdapter.format(date, "DD MMM YYYY") : date}`,
+      );
+      if (errors?.length) {
+        console.log(
+          `Error(s): ${errors
+            .map(({ type, message }) => `type=${type} message=${message}`)
+            .join(",")}`,
+        );
+        if (value) {
+          console.log(`Original Value: ${value}`);
+        }
+      }
+    },
+    [dateAdapter],
+  );
+
+  return (
+    <StackLayout>
+      <StackLayout direction={"row"}>
+        <FormField>
+          <FormFieldLabel>Number of columns</FormFieldLabel>
+          <Input
```

Comment:
Maybe number input or `type="number"` could be good here for these examples with arrow up/down support

Reply:
@tayyabataimur  one for your radar